### DISCUSSION TITLE
Is there a way to (pre) filter rofi list

### DISCUSSION THREAD

**Discussion by wolfisdev** [davatorium/rofi#1776] [Q&A]
What I'm trying to achieve is to filter a rofi list before its shown to the user. Specifically: I'm using rofi within qtile as a window switcher / launcher with the combi mode. I have some scratchpad windows that I'd prefer to not show up in the rofi list. I can mark these with a specific value in the group/desktop or wm_class names but I don't know how to exclude these from the rofi list.

What achieves the desired result list is

`rofi -show combi -filter -scratchpad`

except that I'd prefer to not have the `-scratchpad` filter pretyped in the rofi input filter.

Can anyone help me?

Thanks in advance, cheers!

**Comment by DaveDavenport** [comment]
No this is not possible directly in rofi outside of what you found.

You can always write your own window mode for qtile via f.e. script-mode or a plugin.
